app/models/article.rb

```diff
@@ -126,13 +127,16 @@ def update_or_create_redirect
   end
 
   def update_locale_articles_count
-    return if locale.blank?
+    original_locale = saved_change_to_locale.first if saved_change_to_locale?
+    return if locale.blank? && original_locale.blank?
 
-    article_locale = Locale.find_by(abbreviation: locale)
+    article_locales = Locale.where(abbreviation: [locale, original_locale].compact)
 
-    return if article_locale.blank?
+    return if article_locales.blank?
 
-    articles_count = Article.live.published.where(locale: article_locale.abbreviation).count
-    article_locale.update(articles_count: articles_count)
+    article_locales.each do |article_locale|
+      articles_count = Article.live.published.where(locale: article_locale.abbreviation).count
+      article_locale.update(articles_count: articles_count)
```

Comment:
Is there a reason why we keep computed data in table instead of computing them on the fly? It'd avoid all this update / delete logic, tho?

Reply:
I assume it is because there would be a large number of database calls on the /languages page. perhaps partials could be cached at the view layer, but that is a larger refactor and I am not sure it would even be a better situation.

I lean towards moving this logic into a background job, that has a unique constraint, to isolate the logic and testing

Reply:
Ok, for now let's merge it that way, but let's move it to a background job or find a way to avoid this logic in another PR.